subxt/src/backend/chain_head/mod.rs

```diff
@@ -84,11 +86,35 @@ impl<T: Config> ChainHeadBackendBuilder<T> {
     /// give up waiting.
     ///
     /// Provide a value here to denote how long, in seconds, to wait before giving up. Defaults to 240 seconds.
+    ///
+    /// If [`Self::submit_transactions_ignoring_follow_events()`] is called, this timeout is ignored.
     pub fn transaction_timeout(mut self, timeout_secs: usize) -> Self {
         self.transaction_timeout_secs = timeout_secs;
         self
     }
 
+    /// When a transaction is submitted, we normally synchronize the events that we get back with events from
+    /// our background `chainHead_follow` subscription, to ensure that any blocks hashes that we see can be
+    /// immediately queried (for example to get events or state at that block), and are kept around unless they
+    /// are no longer needed.
+    ///
+    /// The main downside of this synchronization is that there may be a delay in being handed back a
+    /// [`TransactionStatus::InFinalizedBlock`] event while we wait to see the same block hash emitted from
+    /// our background `chainHead_follow` subscription in order to ensure it's available for querying.
+    ///
+    /// Calling this method turns off this synchronization, speeding up the response and removing any reliance
+    /// on the `chainHead_follow` subscription continuing to run without stopping throughout submitting a transaction.
+    ///
+    /// # Warning
+    ///
+    /// This can lead to errors when calling APIs like `wait_for_finalized_success`, which will try to retrieve events
+    /// at the finalized block, because there will be a race and the finalized block may not be available for querying
+    /// yet.
+    pub fn submit_transactions_ignoring_follow_events(mut self) -> Self {
```

Comment:
a little long name :)

I don't have a better suggestion in mind though

Reply:
Yeah; it's quite long but also quite a rare thing to set I would hope, so I don't mind I guess!